test/interface.spec.js

```diff
@@ -0,0 +1,179 @@
+/* eslint-env mocha */
+'use strict'
+
+const tests = require('interface-ipfs-core')
+const isNode = require('detect-node')
+const CommonFactory = require('./utils/interface-common-factory')
+const IPFSApi = require('../src')
+const isWindows = process.platform && process.platform === 'win32'
+
+describe('interface-ipfs-core tests', () => {
+  const defaultCommonFactory = CommonFactory.create()
+
+  tests.block(defaultCommonFactory)
+
+  tests.bootstrap(defaultCommonFactory)
+
+  tests.config(defaultCommonFactory, {
+    skip: [
+      // config.replace
+      // FIXME Waiting for fix on go-ipfs
+      // - https://github.com/ipfs/js-ipfs-api/pull/307#discussion_r69281789
+      // - https://github.com/ipfs/go-ipfs/issues/2927
+      'replace'
+    ]
+  })
+
+  tests.dag(defaultCommonFactory, {
+    skip: [
+      // dag.tree
+      // TODO vmx 2018-02-22: Currently the tree API is not exposed in go-ipfs
+      'tree',
+      // dag.get:
+      // FIXME vmx 2018-02-22: Currently not supported in go-ipfs, it might
+      // be possible once https://github.com/ipfs/go-ipfs/issues/4728 is
+      // done
+      'should get a dag-pb node local value',
+      'should get dag-pb value via dag-cbor node',
+      'should get by CID string + path',
+      // dag.put
+      // FIXME This works in go-ipfs because dag-pb will serialize any object. If
+      // the object has neither a `data` nor `links` field it's serialized
+      // as an empty object
+      'should not put dag-cbor node with wrong multicodec'
+    ]
+  })
```

Comment:
Hm... is this really the preferred approach? I foresee people changing the test title and not understanding why things are failing. I kind of liked when the tests explicitly communicated that it was known it wasn't working with go-ipfs or js-ipfs.

Reply:
Also worth mentioning - we already have a scattering of where this information is held. We have implementation specific skips in interface-ipfs-core (go/js), but platform dependent (browser/node.js) skips in js-ipfs.

For example, some suites don't work in the browser so aren't included https://github.com/ipfs/js-ipfs/blob/2e9418465c863b129300ea5f0c8b32590edc7729/test/core/interface/interface.spec.js#L17-L23 and we have the same thing for functionality that isn't implemented e.g. repo tests simply just not included in js-ipfs because it's not all implemented yet.

These PRs would put all of that information in the same place, which I think is a win.

I'd like to at least try things this way around and if it becomes an issue then we change it back?

What do you think @diasdavid - is worth a try? I'd like to move forwards on these PRs soon because they're becoming difficult to keep up to date with master!

Thank you, much ❤️

Reply:
I would say Option 1 given the example but since some test titles are actually long, Option 2 might serve us better.

Reply:
@diasdavid is it nice?:

**skip specific test**

<img width="637" alt="screen shot 2018-06-26 at 23 08 11" src="https://user-images.githubusercontent.com/152863/41942074-02348982-7996-11e8-93ce-cc2559b36091.png">

**skip entire subsystem**

<img width="516" alt="screen shot 2018-06-26 at 23 07 26" src="https://user-images.githubusercontent.com/152863/41942084-0cfc74ce-7996-11e8-87a4-8c230b060ec1.png">